contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -558,6 +569,20 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         return false;
     }
 
+    /**
+     * @dev If not already set, initialize blockchainID by getting the current blockchain ID
+     * value from the Warp precompile.
+     * @return The current blockchain ID.
+     */
+    function _initializeBlockchainID() private returns (bytes32) {
```

Comment:
we initialized move the `blockchainID` initialization out of constructor for deployment reasons, but now that we call initialize on a bunch of places, what is the difference with setting it in the state variable declaration again?. i.e. 
`bytes32 public immutable blockchainID = WARP_MESSENGER.getBlockchainID();`

Reply:
The state variable would still be initialized during deployment, so the same consideration around `WARP_MESSENGER.getBlockchainID` reverting would still apply.